Firestore/core/src/firebase/firestore/util/hard_assert.h

```diff
@@ -131,6 +131,8 @@ ABSL_ATTRIBUTE_NORETURN void DefaultFailureHandler(const char* file,
 
 namespace internal {
 
+extern FailureHandler failure_handler_callback;
```

Comment:
Rather than exposing this global, you can make `SetFailureHandler` return the old value.